packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -22,18 +22,39 @@ const { OrgAuthInfo, channelService } = sfdxCoreExports;
 // This should be exported from soql-builder-ui
 export interface SoqlEditorEvent {
   type: string;
-  message?: string;
+  payload?: string | string[] | ToolingModelJson;
 }
 
 // This should be shared with soql-builder-ui
 export enum MessageType {
-  ACTIVATED = 'activated',
-  QUERY = 'query',
+  UI_ACTIVATED = 'ui_activated',
+  UI_SOQL_CHANGED = 'ui_soql_changed',
   SOBJECT_METADATA_REQUEST = 'sobject_metadata_request',
   SOBJECT_METADATA_RESPONSE = 'sobject_metadata_response',
   SOBJECTS_REQUEST = 'sobjects_request',
   SOBJECTS_RESPONSE = 'sobjects_response',
-  UPDATE = 'update'
+  TEXT_SOQL_CHANGED = 'text_soql_changed'
+}
+
+// TODO: move to shared module
+async function getConnection(): Promise<Connection> {
+  const usernameOrAlias = await OrgAuthInfo.getDefaultUsernameOrAlias(true);
+  if (!usernameOrAlias) {
+    // TODO: NLS
+    throw new Error(
+      'No default org is set. Run "SFDX: Create a Default Scratch Org" or "SFDX: Authorize an Org" to set one.'
+    );
+  }
+  return await OrgAuthInfo.getConnection(usernameOrAlias);
+}
+
+async function withSFConnection(f: (conn: Connection) => void): Promise<void> {
+  try {
+    const conn = await getConnection();
+    f(conn);
+  } catch (e) {
+    channelService.appendLine(e);
```

Comment:
Ideally we'd wish to have some telemetry here.

Reply:
Yes.  We are now just working aggressively towards an MVP and plan on swinging back through for telemetry and localization.  It's been prioritized like that for us.  

Future work item is:
W-8086238 - [soql-builder] Add telemetry to get insight into soql builder usage / errors

Reply:
If I understood correctly, if we do what @brpowell  suggests, we might not event need the `catch`

Reply:
I've made the change suggested by @brpowell, but we might still need that catch anyway